docs/developer/architecture/authentication.rst

```diff
@@ -0,0 +1,88 @@
+Authentication Architecture Document
+====================================
+
+Context:
+--------
+Currently, when we deploy MetalK8s we pre provision super admin user with admin
+password. Meaning anyone wanting to use K8S/Salt APIs should authenticate using
```

Comment:
One can trivially add more users as well, with whatever group mappings, and set up custom `(Cluster)RoleBinding`s, so this isn't entirely true.

Reply:
updated in last commit